### PR TITLE
Restructure project to use rollup instead of tsdx

### PR DESCRIPTION
**Pull request by bsunderhus**

### CODE REVIEW

File: jest.config.json [hunk 1]
```diff
@@ -3,5 +3,5 @@
     "testEnvironment": "jest-environment-puppeteer",
     "transform": { "^.+\\.(ts|tsx|js|jsx)?$": "ts-jest" },
     "testPathIgnorePatterns": ["test-utils"],
-    "rootDir": "src"
+    "rootDir": "tests"
```
Comment (ling1726) [inline jest.config.json:6]:
```suggestion
    "rootDir": "src"
```

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        { file: pkg.module, format: "es", sourcemap: true },
+    ],
+    plugins: [
+        json(),
+        replace({
+            preventAssignment: true,
+            __DEV__: `process.env.NODE_ENV === 'development'`,
+            __VERSION__: JSON.stringify(pkg.version),
+        }),
+        // typescript({ tsconfig: "./src/tsconfig.json" }),
```
Comment (ling1726) [inline rollup.config.js:29]:
```suggestion
```

**Comment by ling1726** [comment]
We no longer have separate dev and prod cjs builds... but that's not necessarily a bad thing

**Comment by ling1726** [comment]
Can also fix this: https://github.com/microsoft/tabster/blob/6968ac0284fc4880fa258fb4c41283edec647883/src/Utils.ts#L756

since we're tsdx in no longer in this PR

**Review by ling1726 — DISMISSED** [review]
Let's hold off on this for now until we get the same bundle-size numbers as with tsdx

File: package.json [hunk 3]
```diff
@@ -34,17 +35,23 @@
...
     "dependencies": {
         "keyborg": "^1.0.7",
-        "tslib": "^2.1.0"
+        "tslib": "^2.3.1"
     },
     "devDependencies": {
         "@azure/data-tables": "^13.0.0",
         "@babel/core": "^7.16.0",
-        "@babel/plugin-proposal-class-properties": "^7.16.0",
+        "@babel/plugin-external-helpers": "^7.16.0",
```
Comment (ling1726) [inline package.json:43]: This is never used

File: babel.config.json [hunk 4]
```diff
@@ -1,12 +1,16 @@
 {
-    "presets": ["@babel/preset-react", "@babel/preset-typescript"],
-    "plugins": [
-        "@babel/plugin-proposal-class-properties",
-        "@babel/plugin-proposal-optional-chaining",
-        "@babel/plugin-transform-typescript",
+    "presets": [
```
Comment (ling1726) [inline babel.config.json:2]: Please use the `babel.config.js` in master

File: .browserslistrc [hunk 5]
```diff
@@ -0,0 +1 @@
+IE 11
```
Comment (ling1726) [inline .browserslistrc:1]: we're starting to pick up a log of random config files on repo root. it's not a bad thing but some of them like this, jest, prettier can all be put to package.json

What are your thoughts on that ?

File: .gitignore [hunk 6]
```diff
@@ -19,4 +19,5 @@ storybook-static
 .vs
 
 /dist
-/.rts2*
\ No newline at end of file
+/.rts2*
+*.tsbuildinfo
```
Comment (ling1726) [inline .gitignore:23]: let's remove this for now, since this is for project refs right ?

File: babel.config.js [hunk 7]
```diff
@@ -1,25 +1,25 @@
 module.exports = (api) => {
     const isTest = api.env("test");
 
-    // Currently babel is only used to transform tests
-    if (!isTest) {
-        return {};
-    }
+    const preset = isTest
```
Comment (ling1726) [inline babel.config.js:4]:
```suggestion
    const presetEnv = isTest
```

File: babel.config.js [hunk 8]
```diff
@@ -1,25 +1,25 @@
...
+              },
+          ];
 
     return {
-        presets: [
-            "@babel/preset-react",
-            "@babel/preset-typescript",
-            ["@babel/preset-env", { targets: { node: "current" } }],
-        ],
+        presets: ["@babel/preset-typescript", preset],
```
Comment (ling1726) [inline babel.config.js:16]:
```suggestion
        presets: ["@babel/preset-typescript", presetEnv],
```

File: package-lock.json [hunk 9]
```diff
@@ -3258,10 +3284,191 @@
             }
         },
         "@storybook/addon-measure": {
-            "version": "2.0.0",
-            "resolved": "https://registry.npmjs.org/@storybook/addon-measure/-/addon-measure-2.0.0.tgz",
-            "integrity": "sha512-ZhdT++cX+L9LwjhGYggvYUUVQH/MGn2rwbrAwCMzA/f2QTFvkjxzX8nDgMxIhaLCDC+gHIxfJG2wrWN0jkBr3g==",
-            "dev": true
+            "version": "6.4.0-alpha.27",
```
Comment (ling1726) [inline package-lock.json:3287]: what happened here ? why are there a bunch of storybook dependencies updated ?

File: package-lock.json [hunk 10]
```diff
@@ -19431,12 +19690,63 @@
             "integrity": "sha512-3IVX4nI6B5cc31/GFFE+i8ey/N2eA0CZDbo6n0yrz0zDX8ZJ8djmU1p+XRz7G3is0F3bB3pu2pAroFdAWQKU3w==",
             "dev": true
         },
+        "ts-jest": {
```
Comment (ling1726) [inline package-lock.json:19693]: ts-jest is back ??!!!??? 😱😱😱

File: package.json [hunk 11]
```diff
@@ -46,9 +47,16 @@
...
+        "@rollup/plugin-babel": "^5.3.0",
+        "@rollup/plugin-commonjs": "^21.0.1",
+        "@rollup/plugin-json": "^4.1.0",
+        "@rollup/plugin-node-resolve": "^13.0.6",
+        "@rollup/plugin-replace": "^3.0.0",
+        "@rollup/plugin-typescript": "^8.3.0",
         "@storybook/addon-actions": "^6.3.12",
         "@storybook/addon-essentials": "^6.3.12",
         "@storybook/addon-links": "^6.3.12",
+        "@storybook/addon-measure": "^6.4.0-alpha.27",
```
Comment (ling1726) [inline package.json:59]: why is this addon necessary ?

File: package.json [hunk 12]
```diff
@@ -69,6 +79,10 @@
         "puppeteer": "^5.5.0",
         "requirejs": "^2.3.6",
         "rimraf": "^3.0.2",
+        "rollup": "^2.60.1",
+        "rollup-plugin-sourcemaps": "^0.6.3",
+        "rollup-plugin-typescript2": "^0.31.0",
+        "ts-jest": "^26.4.4",
```
Comment (ling1726) [inline package.json:85]: why is `ts-jest` back ?

File: rollup.config.js [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * @type {import('rollup').RollupOptions}
+ */
+const config = {
+    input: "./src/index.ts",
+    output: [
+        { file: pkg.main, format: "cjs", sourcemap: true },
+        { file: pkg.module, format: "es", sourcemap: true },
+    ],
+    external: ["tslib", "core-js", "keyborg"],
```
Comment (ling1726) [inline rollup.config.js:22]: does core-js need to be there ? as we said we shouldn't use before we kill IE11 support

```suggestion
    external: ["tslib", "keyborg"],
```

File: .browserslistrc (hunk 5 above)
Comment (bsunderhus) [reply .browserslistrc:1]: I agree on having them on `package.json` to clean up the repo. But let's make a PR for doing that all in one

**Review by ling1726 — APPROVED** [review]

**Pull request merged**